lib/KTable/index.vue

```diff
@@ -0,0 +1,484 @@
+  import useSorting, { SORT_ORDER_ASC, SORT_ORDER_DESC, DATA_TYPE_OTHERS } from './useSorting';
+  import KTableGridItem from './KTableGridItem.vue';
+
+  export default {
+    name: 'KTable',
+    components: {
+      KTableGridItem,
+    },
+    setup(props, { emit }) {
+      const headers = ref(props.headers);
+      const rows = ref(props.rows);
+      const useLocalSorting = ref(props.useLocalSorting);
+      const {
+        sortKey,
+        sortOrder,
+        sortedRows,
+        handleSort: localHandleSort,
+        getAriaSort,
+      } = useSorting(headers, rows, useLocalSorting);
+
+      const finalRows = computed(() => {
+        if (props.sortable) {
+          return useLocalSorting.value ? sortedRows.value : rows.value;
+        } else {
+          return rows.value;
+        }
+      });
+
+      watch(
+        () => props.rows,
+        newRows => {
+          rows.value = newRows;
+        }
+      );
+
+      const handleSort = index => {
+        if (headers.value[index].dataType === DATA_TYPE_OTHERS) {
+          return;
+        }
+        if (useLocalSorting.value) {
+          localHandleSort(index);
+        } else {
+          emit(
+            'changeSort',
+            index,
+            sortOrder.value === SORT_ORDER_ASC ? SORT_ORDER_DESC : SORT_ORDER_ASC
+          );
+        }
+      };
+
+      const getHeaderStyle = header => {
+        const style = {};
+        if (header.minWidth) style.minWidth = header.minWidth;
+        if (header.width) style.width = header.width;
+        return style;
+      };
+
+      return {
+        sortKey,
+        sortOrder,
+        finalRows,
+        handleSort,
+        getAriaSort,
+        SORT_ORDER_ASC,
+        SORT_ORDER_DESC,
+        DATA_TYPE_OTHERS,
+        getHeaderStyle,
+      };
+    },
+    /* eslint-disable kolibri/vue-no-unused-properties */
+    props: {
+      /**
+       * An array of objects `{ label, dataType, minWidth, width }`representing the headers of the table. The `dataType` can be one of `'string'`, `'numeric'`, `'date'`, or `'others'`. `label` and `dataType` are required. `minWidth` and `width` are optional.
+       */
+      headers: {
+        type: Array,
+        required: true,
+        validator: function(value) {
+          return value.every(
+            header =>
+              ['label', 'dataType'].every(key => key in header) &&
+              ['string', 'numeric', 'date', 'others'].includes(header.dataType)
+          );
+        },
+      },
+      /**
+       * An array of arrays representing the rows of the table. Each row should have the same number of elements as the headers array.
+       */
+      rows: {
+        type: Array,
+        required: true,
+      },
+      /**
+       * The caption of the table
+       */
+      caption: {
+        type: String,
+        required: true,
+      },
+      /**
+       * Enables or disables local sorting of the table data.
+       */
+      useLocalSorting: {
```

Comment:
While reading the code I noticed this useLocalSorting refers to if it is sorted in the backend (or externally) or in the frontend. I didnt get it by just reading the props section, I think we could specify more what does local sorting refers to, since as a external user that hasnt been involved in the project I got a little bit confused.

Reply:
Note that this prop may be subject to change and hopefully it helps to address this as well https://github.com/learningequality/kolibri-design-system/pull/727#issuecomment-2295168580